src/seedu/addressbook/commands/AddListCommand.java

```diff
@@ -0,0 +1,76 @@
+package seedu.addressbook.commands;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+
+/*
+Add a person to the addressbook and list down everyone in the addressbook right after
+ */
+public class AddListCommand extends Command{
+    public static final String COMMAND_WORD = "add&list";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n" + "Adds a person to the address book and list the whole address book after a successful add. "
+            + "Contact details can be marked private by prepending 'p' to the prefix.\n\t"
+            + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n\t"
+            + "Example: " + COMMAND_WORD
+            + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+
+    //    public static final String MESSAGE_SUCCESS = "New person added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+
+    private final Person toAdd;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public AddListCommand(String name,
+                          String phone, boolean isPhonePrivate,
+                          String email, boolean isEmailPrivate,
+                          String address, boolean isAddressPrivate,
+                          Set<String> tags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toAdd = new Person(
+                new Name(name),
+                new Phone(phone, isPhonePrivate),
+                new Email(email, isEmailPrivate),
+                new Address(address, isAddressPrivate),
+                tagSet
+        );
+    }
+
+    public AddListCommand(Person toAdd) {
+        this.toAdd = toAdd;
+    }
+
+    public ReadOnlyPerson getPerson() {
+        return toAdd;
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            addressBook.addPerson(toAdd);
+            List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+            return new CommandResult(getMessageForPersonListShownSummary(allPersons), allPersons);
+//            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
```

Comment:
Again, avoid leaving behind commented out code for submission. This reduces code readability.